src/Fullcount.sol

```diff
@@ -177,28 +193,45 @@ contract Fullcount is EIP712 {
 
     // Emits:
     // - SessionJoined
-    function joinSession(uint256 sessionID, address nftAddress, uint256 tokenID) external virtual {
+    function joinSession(
+        uint256 sessionID,
+        address nftAddress,
+        uint256 tokenID,
+        bytes memory signature
+    )
+        external
+        virtual
+    {
         require(sessionID <= NumSessions, "Fullcount.joinSession: session does not exist");
 
         require(_isTokenOwner(nftAddress, tokenID), "Fullcount.joinSession: msg.sender is not NFT owner");
 
         require(StakedSession[nftAddress][tokenID] == 0, "Fullcount.joinSession: NFT is already staked to a session.");
 
         Session storage session = SessionState[sessionID];
-        if (session.pitcherNFT.nftAddress != address(0) && session.batterNFT.nftAddress != address(0)) {
+
+        if (session.requireSignature) {
+            bytes32 sessionMessageHash = sessionHash(sessionID);
+            require(
+                SignatureChecker.isValidSignatureNow(session.sessionStarter, sessionMessageHash, signature),
```

Comment:
Why do you need this?

You can look up the `ownerOf` the pitcher (if `session.pitcher.nft.nftAddress != address(0)`) or the batter (if `session.batter.nft.nftAddress != address(0)`).

If both are zero, it means the session was aborted.

Reply:
This has the benefit that, if the ownership of the starting NFT is changed while the starting NFT is waiting for someone to join, then people who had the invitation before would not be able to join. And people who get an invitation after the transfer *would* be able to join.